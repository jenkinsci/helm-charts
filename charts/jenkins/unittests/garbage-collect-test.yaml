suite: Garbage collection of pods
release:
  name: my-release
  namespace: my-namespace
templates:
  - jcasc-config.yaml
tests:
  - it: one cloud
    set:
      agent.garbageCollection.enabled: true
    asserts:
      - isKind:
          of: ConfigMap
      - hasDocuments:
          count: 1
      - isNotEmpty:
          path: data["jcasc-default-config.yaml"]
      - matchRegex:
          path: metadata.labels["helm.sh/chart"]
          pattern: ^jenkins-
      - matchSnapshot:
          path: data["jcasc-default-config.yaml"]
      - equal:
          path: metadata.namespace
          value: my-namespace

  - it: second cloud
    set:
      agent.garbageCollection.enabled: true
      additionalClouds.second.agent.containerCap: 5
    asserts:
      - isKind:
          of: ConfigMap
      - hasDocuments:
          count: 1
      - isNotEmpty:
          path: data["jcasc-default-config.yaml"]
      - matchRegex:
          path: metadata.labels["helm.sh/chart"]
          pattern: ^jenkins-
      - matchSnapshot:
          path: data["jcasc-default-config.yaml"]
      - equal:
          path: metadata.namespace
          value: my-namespace

  - it: namespaces
    set:
      agent.garbageCollection.enabled: true
      agent.garbageCollection.namespaces: |-
        namespace1
        namespace2
    asserts:
      - isKind:
          of: ConfigMap
      - hasDocuments:
          count: 1
      - isNotEmpty:
          path: data["jcasc-default-config.yaml"]
      - matchRegex:
          path: metadata.labels["helm.sh/chart"]
          pattern: ^jenkins-
      - matchSnapshot:
          path: data["jcasc-default-config.yaml"]
      - equal:
          path: metadata.namespace
          value: my-namespace
