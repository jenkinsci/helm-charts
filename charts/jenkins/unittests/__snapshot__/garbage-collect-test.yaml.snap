namespaces:
  1: |
    |-
      jenkins:
        authorizationStrategy:
          loggedInUsersCanDoAnything:
            allowAnonymousRead: false
        securityRealm:
          local:
            allowsSignup: false
            enableCaptcha: false
            users:
            - id: "${chart-admin-username}"
              name: "Jenkins Admin"
              password: "${chart-admin-password}"
        disableRememberMe: false
        mode: NORMAL
        numExecutors: 0
        labelString: ""
        projectNamingStrategy: "standard"
        markupFormatter:
          plainText
        clouds:
        - kubernetes:
            containerCapStr: "10"
            garbageCollection:
              namespaces: |-
                namespace1
                namespace2
              timeout: "300"
            defaultsProviderTemplate: ""
            connectTimeout: "5"
            readTimeout: "15"
            jenkinsUrl: "http://my-release-jenkins.my-namespace.svc.cluster.local:8080"
            jenkinsTunnel: "my-release-jenkins-agent.my-namespace.svc.cluster.local:50000"
            skipTlsVerify: false
            usageRestricted: false
            maxRequestsPerHostStr: "32"
            retentionTimeout: "5"
            waitForPodSec: "600"
            name: "kubernetes"
            namespace: "my-namespace"
            restrictedPssSecurityContext: false
            serverUrl: "https://kubernetes.default"
            credentialsId: ""
            podLabels:
            - key: "jenkins/my-release-jenkins-agent"
              value: "true"
            templates:
              - name: "default"
                namespace: "my-namespace"
                id: d93b785d71723ac24dbdb70e09b810e04c2175d0b8f3b33320dbcdaf2a0f9208
                containers:
                - name: "jnlp"
                  alwaysPullImage: false
                  args: "^${computer.jnlpmac} ^${computer.name}"
                  envVars:
                    - envVar:
                        key: "JENKINS_URL"
                        value: "http://my-release-jenkins.my-namespace.svc.cluster.local:8080/"
                  image: "jenkins/inbound-agent:3309.v27b_9314fd1a_4-3"
                  privileged: "false"
                  resourceLimitCpu: 512m
                  resourceLimitMemory: 512Mi
                  resourceRequestCpu: 512m
                  resourceRequestMemory: 512Mi
                  ttyEnabled: false
                  workingDir: /home/jenkins/agent
                idleMinutes: 0
                instanceCap: 2147483647
                label: "my-release-jenkins-agent "
                nodeUsageMode: "NORMAL"
                podRetention: Never
                showRawYaml: true
                serviceAccount: "default"
                slaveConnectTimeoutStr: "100"
                yamlMergeStrategy: override
                inheritYamlMergeStrategy: false
        slaveAgentPort: 50000
        crumbIssuer:
          standard:
            excludeClientIPFromCrumb: true
      security:
        apiToken:
          creationOfLegacyTokenEnabled: false
          tokenGenerationOnCreationEnabled: false
          usageStatisticsEnabled: true
      unclassified:
        location:
          url: http://my-release-jenkins:8080
one cloud:
  1: |
    |-
      jenkins:
        authorizationStrategy:
          loggedInUsersCanDoAnything:
            allowAnonymousRead: false
        securityRealm:
          local:
            allowsSignup: false
            enableCaptcha: false
            users:
            - id: "${chart-admin-username}"
              name: "Jenkins Admin"
              password: "${chart-admin-password}"
        disableRememberMe: false
        mode: NORMAL
        numExecutors: 0
        labelString: ""
        projectNamingStrategy: "standard"
        markupFormatter:
          plainText
        clouds:
        - kubernetes:
            containerCapStr: "10"
            garbageCollection:
              timeout: "300"
            defaultsProviderTemplate: ""
            connectTimeout: "5"
            readTimeout: "15"
            jenkinsUrl: "http://my-release-jenkins.my-namespace.svc.cluster.local:8080"
            jenkinsTunnel: "my-release-jenkins-agent.my-namespace.svc.cluster.local:50000"
            skipTlsVerify: false
            usageRestricted: false
            maxRequestsPerHostStr: "32"
            retentionTimeout: "5"
            waitForPodSec: "600"
            name: "kubernetes"
            namespace: "my-namespace"
            restrictedPssSecurityContext: false
            serverUrl: "https://kubernetes.default"
            credentialsId: ""
            podLabels:
            - key: "jenkins/my-release-jenkins-agent"
              value: "true"
            templates:
              - name: "default"
                namespace: "my-namespace"
                id: 5ce2c26356ff021297afc00abd3e9707207dcb834f9001e3c200cab5b02dcf7a
                containers:
                - name: "jnlp"
                  alwaysPullImage: false
                  args: "^${computer.jnlpmac} ^${computer.name}"
                  envVars:
                    - envVar:
                        key: "JENKINS_URL"
                        value: "http://my-release-jenkins.my-namespace.svc.cluster.local:8080/"
                  image: "jenkins/inbound-agent:3309.v27b_9314fd1a_4-3"
                  privileged: "false"
                  resourceLimitCpu: 512m
                  resourceLimitMemory: 512Mi
                  resourceRequestCpu: 512m
                  resourceRequestMemory: 512Mi
                  ttyEnabled: false
                  workingDir: /home/jenkins/agent
                idleMinutes: 0
                instanceCap: 2147483647
                label: "my-release-jenkins-agent "
                nodeUsageMode: "NORMAL"
                podRetention: Never
                showRawYaml: true
                serviceAccount: "default"
                slaveConnectTimeoutStr: "100"
                yamlMergeStrategy: override
                inheritYamlMergeStrategy: false
        slaveAgentPort: 50000
        crumbIssuer:
          standard:
            excludeClientIPFromCrumb: true
      security:
        apiToken:
          creationOfLegacyTokenEnabled: false
          tokenGenerationOnCreationEnabled: false
          usageStatisticsEnabled: true
      unclassified:
        location:
          url: http://my-release-jenkins:8080
second cloud:
  1: |
    |-
      jenkins:
        authorizationStrategy:
          loggedInUsersCanDoAnything:
            allowAnonymousRead: false
        securityRealm:
          local:
            allowsSignup: false
            enableCaptcha: false
            users:
            - id: "${chart-admin-username}"
              name: "Jenkins Admin"
              password: "${chart-admin-password}"
        disableRememberMe: false
        mode: NORMAL
        numExecutors: 0
        labelString: ""
        projectNamingStrategy: "standard"
        markupFormatter:
          plainText
        clouds:
        - kubernetes:
            containerCapStr: "10"
            garbageCollection:
              timeout: "300"
            defaultsProviderTemplate: ""
            connectTimeout: "5"
            readTimeout: "15"
            jenkinsUrl: "http://my-release-jenkins.my-namespace.svc.cluster.local:8080"
            jenkinsTunnel: "my-release-jenkins-agent.my-namespace.svc.cluster.local:50000"
            skipTlsVerify: false
            usageRestricted: false
            maxRequestsPerHostStr: "32"
            retentionTimeout: "5"
            waitForPodSec: "600"
            name: "kubernetes"
            namespace: "my-namespace"
            restrictedPssSecurityContext: false
            serverUrl: "https://kubernetes.default"
            credentialsId: ""
            podLabels:
            - key: "jenkins/my-release-jenkins-agent"
              value: "true"
            templates:
              - name: "default"
                namespace: "my-namespace"
                id: 5ce2c26356ff021297afc00abd3e9707207dcb834f9001e3c200cab5b02dcf7a
                containers:
                - name: "jnlp"
                  alwaysPullImage: false
                  args: "^${computer.jnlpmac} ^${computer.name}"
                  envVars:
                    - envVar:
                        key: "JENKINS_URL"
                        value: "http://my-release-jenkins.my-namespace.svc.cluster.local:8080/"
                  image: "jenkins/inbound-agent:3309.v27b_9314fd1a_4-3"
                  privileged: "false"
                  resourceLimitCpu: 512m
                  resourceLimitMemory: 512Mi
                  resourceRequestCpu: 512m
                  resourceRequestMemory: 512Mi
                  ttyEnabled: false
                  workingDir: /home/jenkins/agent
                idleMinutes: 0
                instanceCap: 2147483647
                label: "my-release-jenkins-agent "
                nodeUsageMode: "NORMAL"
                podRetention: Never
                showRawYaml: true
                serviceAccount: "default"
                slaveConnectTimeoutStr: "100"
                yamlMergeStrategy: override
                inheritYamlMergeStrategy: false
        - kubernetes:
            containerCapStr: "5"
            garbageCollection:
              timeout: "300"
            defaultsProviderTemplate: ""
            connectTimeout: "5"
            readTimeout: "15"
            jenkinsUrl: "http://my-release-jenkins.my-namespace.svc.cluster.local:8080"
            jenkinsTunnel: "my-release-jenkins-agent.my-namespace.svc.cluster.local:50000"
            skipTlsVerify: false
            usageRestricted: false
            maxRequestsPerHostStr: "32"
            retentionTimeout: "5"
            waitForPodSec: "600"
            name: "second"
            namespace: "my-namespace"
            restrictedPssSecurityContext: false
            serverUrl: "https://kubernetes.default"
            credentialsId: ""
            podLabels:
            - key: "jenkins/my-release-jenkins-agent"
              value: "true"
            templates:
              - name: "default"
                namespace: "my-namespace"
                id: 648ac2da01795a0055a4689012336c5cf9c8cb03dfdfa06842084c7fb13aef58
                containers:
                - name: "jnlp"
                  alwaysPullImage: false
                  args: "^${computer.jnlpmac} ^${computer.name}"
                  envVars:
                    - envVar:
                        key: "JENKINS_URL"
                        value: "http://my-release-jenkins.my-namespace.svc.cluster.local:8080/"
                  image: "jenkins/inbound-agent:3309.v27b_9314fd1a_4-3"
                  privileged: "false"
                  resourceLimitCpu: 512m
                  resourceLimitMemory: 512Mi
                  resourceRequestCpu: 512m
                  resourceRequestMemory: 512Mi
                  ttyEnabled: false
                  workingDir: /home/jenkins/agent
                idleMinutes: 0
                instanceCap: 2147483647
                label: "my-release-jenkins-agent "
                nodeUsageMode: "NORMAL"
                podRetention: Never
                showRawYaml: true
                serviceAccount: "default"
                slaveConnectTimeoutStr: "100"
                yamlMergeStrategy: override
                inheritYamlMergeStrategy: false
        slaveAgentPort: 50000
        crumbIssuer:
          standard:
            excludeClientIPFromCrumb: true
      security:
        apiToken:
          creationOfLegacyTokenEnabled: false
          tokenGenerationOnCreationEnabled: false
          usageStatisticsEnabled: true
      unclassified:
        location:
          url: http://my-release-jenkins:8080
